core-bundle/src/Resources/contao/library/Contao/System.php

```diff
@@ -521,6 +522,12 @@ public static function loadLanguageFile($strName, $strLanguage=null, $blnNoCache
 			}
 		}
 
+		// Set MSC.textDirection (see #3360)
+		if ('default' === $strName && empty($GLOBALS['TL_LANG']['MSC']['textDirection']))
+		{
+			$GLOBALS['TL_LANG']['MSC']['textDirection'] = Cosmo::create($strLanguage)->direction();
```

Comment:
```suggestion
			$GLOBALS['TL_LANG']['MSC']['textDirection'] = (\ResourceBundle::create($strLanguage, 'ICUDATA', true)['layout']['characters'] ?? null) === 'right-to-left' ? 'rtl' : 'ltr';
```

I would prefer not using a third party library for that use case.

Reply:
Agreed, though the `create` and `direction` methods of the library also have error handling which would then be missing here.